### PR TITLE
Ts migration

### LINKED ISSUE #1413: Truncate.js
file: https://github.com/RedHatInsights/frontend-components/tree/master/packages/components/src/Truncate/Truncate.js

### LINKED ISSUE #1414: DateFormat.js
file: https://github.com/RedHatInsights/frontend-components/tree/master/packages/components/src/DateFormat/DateFormat.js

### PR DESCRIPTION
**Pull request by Hyperkid123**
closes #1413 
closes #1414

### CODE REVIEW

File: packages/components/src/DateFormat/helper.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        {extraTitle}
+        {date}
+      </div>
+    }
+  >
+    {element}
+  </Tooltip>
+);
+
+export type DateType = 'exact' | 'onlyDate' | 'relative' | 'invalid';
```
Comment (rvsia) [inline packages/components/src/DateFormat/helper.tsx:37]: Could we use this in DateFormatProps (with omitting the invalid), so the types are defined in one place?

File: packages/components/src/Truncate/Truncate.tsx [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
-const dangerousHtml = (html) => ({ __html: sanitizeHtml(html) });
+const dangerousHtml = (html: string) => ({ __html: sanitizeHtml(html) });
 
-const Truncate = ({
+export interface TruncateProps {
+  className?: string;
+  text?: string;
+  length?: number;
+  expandText?: string;
+  collapseText?: string;
```
Comment (rvsia) [inline packages/components/src/Truncate/Truncate.tsx:15]: React node is not supported here?

Comment (rvsia) [reply packages/components/src/Truncate/Truncate.tsx:15]: 😈 🔥 😈

Comment (Hyperkid123) [reply packages/components/src/Truncate/Truncate.tsx]: Done. Rest must be a string.

File: packages/components/src/DateFormat/helper.tsx (hunk 1 above)
Comment (Hyperkid123) [reply packages/components/src/DateFormat/helper.tsx:37]: Exclude might work

Comment (Hyperkid123) [reply packages/components/src/DateFormat/helper.tsx:37]: And it does

**Review by rvsia — APPROVED** [review]

**Pull request merged**